### PR TITLE
Feature/login and history

### PR DESCRIPTION
**Pull request by RomiKMoveo**
I have:
- [ ] deleted all irrelevant comments, logs and debuggers.
- [ ] assure that I don't print full error stacks, full body response or headers from the server in the console.
- [ ] deleted all sensitive information from the code because it's being imported from environment variables.
- [ ] encrypted keys/passwords that are sent through the internet.
- [ ] sanitized all user inputs to prevent attacks.
- [ ] made sure my code is generic and reusable for future things or reused existing resources.
- [ ] made sure that my code is readable.
- [ ] checked that the functions and resource names are self-explanatory.
- [ ] tested the code locally and it works as expected.
- [ ] added validations and error handling in relevant cases (external requests, integrations, etc).

### CODE REVIEW

File: pokemon-minisite/src/app/style/variables.scss [hunk 1]
```diff
@@ -44,3 +44,28 @@
 
     /* FONTS */
     $main-font: 'apple-system', 'BlinkMacSystemFont', 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Open Sans', 'Helvetica Neue', sans-serif;
+
+    @mixin btn-type-button {
+        font-family: sans-serif;
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/style/variables.scss]: use font as variable

File: pokemon-minisite/src/app/style/variables.scss [hunk 2]
```diff
@@ -44,3 +44,28 @@
 
     /* FONTS */
     $main-font: 'apple-system', 'BlinkMacSystemFont', 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Open Sans', 'Helvetica Neue', sans-serif;
+
+    @mixin btn-type-button {
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/style/variables.scss]: add the mixin to another file - mixim.scss in the style folder

File: pokemon-minisite/src/app/app.component.ts [hunk 3]
```diff
@@ -5,10 +5,11 @@ import { PokemonListComponent } from './components/pokemon-list/pokemon-list.com
 import { HeaderComponent } from './components/header/header.component';
 
 
+
 @Component({
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/app.component.ts:9]: you don't need the imports here. The imports needs to be in the module of the component

File: pokemon-minisite/src/app/components/header/header.component.ts [hunk 4]
```diff
@@ -1,13 +1,47 @@
...
-export class HeaderComponent {
+export class HeaderComponent implements OnInit {
+  isVerifiedSubject: Observable<boolean>;
+  
+  constructor(
+    private router: Router,
+    private loginService: LoginService
+  ) 
+  {
+    this.isVerifiedSubject = this.loginService.getisVerifiedSubjectObservable();
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/header/header.component.ts]: its unnesecery

File: pokemon-minisite/src/app/services/login.service.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import { BehaviorSubject, Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LoginService {
+  private isVerifiedSubject: BehaviorSubject<boolean>;
+
+  constructor() {
+    const isVerifiedLS = localStorage.getItem('isVerified') === 'true';
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/services/login.service.ts]: Change to a more meaningful name
Don't use it in the constructor.
Take it to the function

File: pokemon-minisite/src/app/services/login.service.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    this.isVerifiedSubject = new BehaviorSubject<boolean>(isVerifiedLS);
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/services/login.service.ts]: why you don't use the set /function?

File: pokemon-minisite/src/app/components/header/header.component.ts [hunk 7]
```diff
@@ -1,13 +1,47 @@
...
-  styleUrl: './header.component.scss'
+  styleUrl: './header.component.scss',
+  imports: [
+    CommonModule,
+  
+  ]
 })
-export class HeaderComponent {
+export class HeaderComponent implements OnInit {
+  isVerifiedSubject: Observable<boolean>;
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/header/header.component.ts]: why this var is behavior subject?

File: pokemon-minisite/src/app/components/header/header.component.ts [hunk 8]
```diff
@@ -1,13 +1,47 @@
...
+    private router: Router,
+    private loginService: LoginService
+  ) 
+  {
+    this.isVerifiedSubject = this.loginService.getisVerifiedSubjectObservable();
+  }
+  
+  ngOnInit(): void {
+    this.loginService.getisVerifiedSubjectObservable().subscribe(() => {
+      this.isVerifiedSubject = this.loginService.getisVerifiedSubjectObservable();
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/header/header.component.ts]: why you don't check just the value?

File: pokemon-minisite/src/app/components/header/header.component.ts [hunk 9]
```diff
@@ -1,13 +1,47 @@
...
+  });
+  }
+    
+
+  logout() {
+    this.loginService.setisVerifiedSubject(false);
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/header/header.component.ts]: you can delete from local

File: pokemon-minisite/src/app/components/header/header.component.html [hunk 10]
```diff
@@ -1,6 +1,8 @@
 <div class="header">
-    <header></header>
+  <div *ngIf="isVerifiedSubject | async">
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/header/header.component.html]: use the if in the app component

File: pokemon-minisite/src/app/components/header/header.component.html [hunk 11]
```diff
@@ -1,6 +1,8 @@
...
+    <button type="button" class="btn" (click)="logout()">Logout</button>
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/header/header.component.html]: try to remove the type button

File: pokemon-minisite/src/environment.ts [hunk 12]
```diff
@@ -0,0 +1,6 @@
+export const environment = {
+    production: false,
+    apiUrl: 'https://pokeapi.co/api/v2/pokemon/',
+    imageUrl: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/',
```
Comment (noyflaysher) [inline pokemon-minisite/src/environment.ts:4]: enter the image and verify mail to params object

File: pokemon-minisite/src/app/components/login/login.component.ts [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+  readonly verifyEmail: string = environment.verifyEmail;
+  errorMessage: string | undefined;
+
+  
+  constructor(
+    private router: Router,
+    private loginService: LoginService,
+  ){}
+
+  ngOnInit(): void {
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/login/login.component.ts]: remove if you don't use

File: pokemon-minisite/src/app/components/login/login.component.scss [hunk 14]
```diff
@@ -0,0 +1,89 @@
+@use '../../style/variables' as *;
+body {
+  font-family: $main-font;
+  height: 100vh;
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+  background-image: url("https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/5bbf58a6-1e85-4cfe-83fd-02df6f482b45/de8nlib-6bed7b3d-3d7e-4763-bb60-18f5ee0127fd.png/v1/fill/w_1024,h_652,q_80,strp/background_prairie_pokemon_screencapture_by_nemotrex_de8nlib-fullview.jpg?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7ImhlaWdodCI6Ijw9NjUyIiwicGF0aCI6IlwvZlwvNWJiZjU4YTYtMWU4NS00Y2ZlLTgzZmQtMDJkZjZmNDgyYjQ1XC9kZThubGliLTZiZWQ3YjNkLTNkN2UtNDc2My1iYjYwLTE4ZjVlZTAxMjdmZC5wbmciLCJ3aWR0aCI6Ijw9MTAyNCJ9XV0sImF1ZCI6WyJ1cm46c2VydmljZTppbWFnZS5vcGVyYXRpb25zIl19.ZVgNGi61AOkC068E-wphWmAUW8amj0MiJySqDcMCVh8");
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/login/login.component.scss]: save the img in the project

File: pokemon-minisite/src/app/components/login/login.component.scss [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+  background-size: cover; 
+  background-position: center;
+ 
+  .box {
+      background-color: rgba(0, 0, 0, 0.8);
+      border-radius: 10px;
+      box-shadow: 0 20px 30px rgba(0, 0, 0, 0.8);
+      margin: auto auto;
+      padding: 40px;
+      position: absolute;
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/login/login.component.scss:20]: Try to center it without absolute

File: pokemon-minisite/src/app/components/login/login.component.html [hunk 16]
```diff
@@ -0,0 +1,16 @@
+<body>
+  <main class="box">
+      <h2>Login</h2>
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/login/login.component.html]: Use div and give it style. without h2...

File: pokemon-minisite/src/app/components/login/login.component.html [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+      <div>
+        <div class="inputBox">
+          <h4 *ngIf="errorMessage" class="error-message">{{ errorMessage }}</h4> 
+          <label for="userName">Username</label>
+          <input type="email"  [(ngModel)]="email"  placeholder="Type your username" required/>   
+        </div>
+        <div>
+          <button type="submit" name=""  (click)="emailVerification()">Submit</button>
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/login/login.component.html]: remove the name

File: pokemon-minisite/src/app/components/history/history.component.ts [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+import { Router } from '@angular/router';
+import { LoginService } from '../../services/login.service';
+import { PokemonListComponent } from '../pokemon-list/pokemon-list.component';
+
+@Component({
+  selector: 'app-history',
+  standalone: true,
+  imports: [
+    NgFor,
+    FormsModule,
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/history/history.component.ts:14]: import in the module

File: pokemon-minisite/src/app/components/history/history.component.html [hunk 19]
```diff
@@ -0,0 +1,17 @@
+<body>
+  <ng-container *ngIf="searchHistory.length > 0; else noHistory">
+    <main class="box">
+      <h2>Search History</h2>
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/history/history.component.html]: dont use h2

File: pokemon-minisite/src/app/components/history/history.component.html [hunk 20]
```diff
@@ -0,0 +1,17 @@
...
+      <div class="inputBox">
+        <div *ngFor="let history of searchHistory; let i = index">
+          <h3 for="history">{{ i + 1 }}. {{ history }}</h3>
+        </div>
+      </div>
+        <button type="button" name="" (click)="goBack()">home</button>
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/history/history.component.html]: remove the type and the name

File: pokemon-minisite/src/app/components/history/history.component.scss [hunk 21]
```diff
@@ -0,0 +1,73 @@
+@use '../../style/variables' as *;
+
+body {
+    font-family: $main-font;
+    height: 100vh;
+    margin: 0;
+    padding: 0;
+    box-sizing: border-box;
+    background-image: url("https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/5bbf58a6-1e85-4cfe-83fd-02df6f482b45/de8nlib-6bed7b3d-3d7e-4763-bb60-18f5ee0127fd.png/v1/fill/w_1024,h_652,q_80,strp/background_prairie_pokemon_screencapture_by_nemotrex_de8nlib-fullview.jpg?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7ImhlaWdodCI6Ijw9NjUyIiwicGF0aCI6IlwvZlwvNWJiZjU4YTYtMWU4NS00Y2ZlLTgzZmQtMDJkZjZmNDgyYjQ1XC9kZThubGliLTZiZWQ3YjNkLTNkN2UtNDc2My1iYjYwLTE4ZjVlZTAxMjdmZC5wbmciLCJ3aWR0aCI6Ijw9MTAyNCJ9XV0sImF1ZCI6WyJ1cm46c2VydmljZTppbWFnZS5vcGVyYXRpb25zIl19.ZVgNGi61AOkC068E-wphWmAUW8amj0MiJySqDcMCVh8");
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/history/history.component.scss]: save the img in the project

File: pokemon-minisite/src/app/components/history/history.component.scss [hunk 22]
```diff
@@ -0,0 +1,73 @@
+@use '../../style/variables' as *;
+
+body {
+    font-family: $main-font;
+    height: 100vh;
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/history/history.component.scss]: use %

File: pokemon-minisite/src/app/components/history/history.component.scss [hunk 23]
```diff
@@ -0,0 +1,73 @@
...
+    padding: 0;
+    box-sizing: border-box;
+    background-image: url("https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/5bbf58a6-1e85-4cfe-83fd-02df6f482b45/de8nlib-6bed7b3d-3d7e-4763-bb60-18f5ee0127fd.png/v1/fill/w_1024,h_652,q_80,strp/background_prairie_pokemon_screencapture_by_nemotrex_de8nlib-fullview.jpg?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7ImhlaWdodCI6Ijw9NjUyIiwicGF0aCI6IlwvZlwvNWJiZjU4YTYtMWU4NS00Y2ZlLTgzZmQtMDJkZjZmNDgyYjQ1XC9kZThubGliLTZiZWQ3YjNkLTNkN2UtNDc2My1iYjYwLTE4ZjVlZTAxMjdmZC5wbmciLCJ3aWR0aCI6Ijw9MTAyNCJ9XV0sImF1ZCI6WyJ1cm46c2VydmljZTppbWFnZS5vcGVyYXRpb25zIl19.ZVgNGi61AOkC068E-wphWmAUW8amj0MiJySqDcMCVh8");
+    background-size: cover; 
+    background-position: center;
+   
+  
+  
+  .box {
+    background-color: rgba(0, 0, 0, 0.8);
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/history/history.component.scss]: use var

File: pokemon-minisite/src/app/components/history/history.component.scss [hunk 24]
```diff
@@ -0,0 +1,73 @@
...
+    border-radius: 10px;
+    box-shadow: 0 20px 30px rgba(0, 0, 0, 0.8);
+    margin: auto auto;
+    padding: 40px;
+    position: absolute;
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/history/history.component.scss]: try to center without absolute

File: pokemon-minisite/src/app/components/history/history.component.scss [hunk 25]
```diff
@@ -0,0 +1,73 @@
...
+    color: $white;
+    letter-spacing: 2px;
+    margin: 0 0 30px 0;
+    outline: none;
+    width: 100%;
+    text-align: center;
+  }
+  
+  input[type="button"], .box button[type="button"], a.button {
+    font-family: sans-serif;
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/history/history.component.scss]: use var

File: pokemon-minisite/src/app/components/history/history.component.scss [hunk 26]
```diff
@@ -0,0 +1,73 @@
...
+    letter-spacing: 2px;
+    outline: none;
+    text-transform: uppercase;
+    text-decoration: none;
+    margin: 2px 10px 2px 0;
+    display: inline-block;
+    font-size: 14px; 
+  }
+  
+  .box input[type="button"]:hover, .box button[type="button"]:hover, a.button:hover {
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/history/history.component.scss]: use just the name of the class

File: pokemon-minisite/src/app/components/history/history.component.scss (hunk 24 above)
Comment (noyflaysher) [reply pokemon-minisite/src/app/components/history/history.component.scss]: if its the same for all the input use mixim

File: pokemon-minisite/src/app/components/history/history.component.scss (hunk 25 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/history/history.component.scss]: DONE

File: pokemon-minisite/src/app/components/history/history.component.scss (hunk 22 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/history/history.component.scss]: done

File: pokemon-minisite/src/app/components/history/history.component.html (hunk 20 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/history/history.component.html]: removed name needed type to stay in order to keep home button style

File: pokemon-minisite/src/app/components/login/login.component.html (hunk 17 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/login/login.component.html]: done

File: pokemon-minisite/src/app/components/header/header.component.html (hunk 11 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/header/header.component.html]: the style won't apply if I removed the type

File: pokemon-minisite/src/app/style/variables.scss (hunk 1 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/style/variables.scss]: done

File: pokemon-minisite/src/app/style/variables.scss (hunk 2 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/style/variables.scss]: done

File: pokemon-minisite/src/environment.ts (hunk 12 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/environment.ts:4]: done

File: pokemon-minisite/src/app/services/login.service.ts (hunk 5 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/services/login.service.ts]: done

File: pokemon-minisite/src/app/components/history/history.component.scss (hunk 24 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/history/history.component.scss]: done

File: pokemon-minisite/src/app/components/history/history.component.scss (hunk 23 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/history/history.component.scss]: done

File: pokemon-minisite/src/app/services/login.service.ts (hunk 6 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/services/login.service.ts]: done

File: pokemon-minisite/src/app/components/header/header.component.ts (hunk 7 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/header/header.component.ts]: done

File: pokemon-minisite/src/app/components/header/header.component.ts (hunk 8 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/header/header.component.ts]: done

File: pokemon-minisite/src/app/components/login/login.component.ts (hunk 13 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/login/login.component.ts]: done

File: pokemon-minisite/src/app/components/login/login.component.scss (hunk 15 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/login/login.component.scss:20]: done

File: pokemon-minisite/src/app/components/history/history.component.scss (hunk 26 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/history/history.component.scss]: done

File: pokemon-minisite/src/app/components/history/history.component.scss (hunk 21 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/history/history.component.scss]: done

File: pokemon-minisite/src/app/components/header/header.component.ts (hunk 4 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/header/header.component.ts]: done

File: pokemon-minisite/src/app/components/login/login.component.scss (hunk 14 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/login/login.component.scss]: done

File: pokemon-minisite/src/app/components/login/login.component.html (hunk 16 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/login/login.component.html]: done

File: pokemon-minisite/src/app/components/history/history.component.html (hunk 19 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/history/history.component.html]: done

**Pull request merged**